js/huobipro.js

```diff
@@ -811,7 +811,7 @@ module.exports = class huobipro extends Exchange {
                 if (price === undefined) {
                     throw new InvalidOrder (this.id + " market buy order requires price argument to calculate cost (total amount of quote currency to spend for buying, amount * price). To switch off this warning exception and specify cost in the amount argument, set .options['createMarketBuyOrderRequiresPrice'] = false. Make sure you know what you're doing.");
                 } else {
-                    request['amount'] = this.priceToPrecision (symbol, parseFloat (amount) * parseFloat (price));
+                    request['amount'] = this.amountToPrecision (symbol, parseFloat (amount) * parseFloat (price));
```

Comment:
Are you sure that this is right? I mean, if they expect the amount to be the full cost of the order (amount * price), then from there it follows, that the value is in quote currency, therefore the price precision should be applied, not the amount precision... Which particular market are you having trouble with? Can you plz post some code to reproduce it or your verbose output for the error?

Reply:
I'm better in code than English )

```
try {
  const buy = await huobi.createOrder('ETH/BTC', 'market', 'buy', 0.01, 0.031855)
  console.log('%j', buy)
catch (err) {
  console.error('%s', err.message)
}
```

```
// With priceToPrecision
huobiru {"status":"error","err-code":"order-orderamount-precision-error","err-msg":"order amount precision error, scale: `4`","data":null}
```

```
// With amountToPrecision
{
  "info": {
    "status": "ok",
    "data": "20559633275"
  },
  "id": "20559633275",
  "timestamp": 1545978772757,
  "datetime": "2018-12-28T06:32:52.757Z",
  "symbol": "ETH/BTC",
  "type": "market",
  "side": "buy",
  "price": 0.031855,
  "amount": 0.01
}
```

Another argument - you can input only 4-digit precision amount on Huobi web site while filling market buy order (I don't now how to paste screenshots)

Reply:
```
await huobi.fetchOrder(20559633275)
```

```
{
  "info": {
    "id": 20559633275,
    "symbol": "ethbtc",
    "account-id": 5685075,
    "amount": "0.000300000000000000",
    "price": "0.0",
    "created-at": 1545978772585,
    "type": "buy-market",
    "field-amount": "0.009400000000000000",
    "field-cash-amount": "0.000299361800000000",
    "field-fees": "0.000018800000000000",
    "finished-at": 1545978772775,
    "source": "spot-api",
    "state": "filled",
    "canceled-at": 0
  },
  "id": "20559633275",
  "timestamp": 1545978772585,
  "datetime": "2018-12-28T06:32:52.585Z",
  "symbol": "ETH/BTC",
  "type": "market",
  "side": "buy",
  "average": 0.031847,
  "cost": 0.0002993618,
  "amount": 0.0094,
  "filled": 0.0094,
  "remaining": 0,
  "status": "closed",
  "fee": {
    "cost": 0.0000188,
    "currency": "ETH"
  }
}
```